array-manipulation-3/src/flatten.js

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-unused-vars */
+
+function flatten(array) {
+  var resultArray = [];
+
+  for (var i = 0; i < array.length; i++) {
+    if (!Array.isArray(array[i])) {
+      resultArray.push(array[i]);
+    }
+
+    for (var x = 0; x < array[i].length; x++) {
+      if (Array.isArray(array[i])) {
+        resultArray.push(array[i][x]);
+      }
+    }
```

Comment:
You should include an `else` in here, otherwise you're checking the `length` of something that probably doesn't have a `length`.